src/pybloomfilter.pyx

```diff
@@ -176,6 +178,8 @@ cdef class BloomFilter:
                 else:
                     cpython.PyErr_NoMemory()
 
+        self._read_only = not (self._in_memory or 'w' in mode)
```

Comment:
For the `mode` string, only 'r' implies read-only. You can use 'a' for appending, or 'r+' for read-write without truncation.

Reply:
I refactored to make it clear that we are only interested in writability.